### PR TITLE
Remove deprecated fields from REST API for 1.0

### PR DESCRIPTION
**Pull request by aarondav**
## What changes are proposed in this pull request?
 
The fields `run_name`, `source_name`, `source_version`, `source_type`, and `entry_point_name` were deprecated in 0.9.1 (moved to tags). This PR removes these fields from the REST API. Additionally, `parent_run_id` has always been a tag in the backend, so we go ahead and remove it from the tracking API.

The changes required to support this removal are widespread throughout every API, including the JavaScript code. Note that we will have to provide migration code to support `source_name`, `source_version`, `source_type`, and `entry_point_name` from runs created prior to 0.9.1 -- as these fields were canonically stored in RunInfo, they will be absent from search_runs and the UI.

Also removed is `anded_expressions`, which has been deprecated since 0.9.0.
 
## How is this patch tested?
 
- [x] Existing unit tests
 
### Is this a user-facing change? 

As deprecated in 0.9.1 and before, the RunInfo fields `run_name`, `source_name`, `source_version`, `source_type`, and `entry_point_name` and the SearchRuns field `anded_expressions` have been removed from the REST API and Python, Java, and R tracking client APIs. They are still available as tags, documented in the REST API documentation.
 
### What component(s) does this PR affect?
 
- [x] UI
- [x] API 
- [x] REST-API 
- [x] Tracking
- [x] R
- [x] Java
- [x] Python

### How should the PR be classified in the release notes? Choose one:
 
- [x] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section

### CODE REVIEW

**Review by smurching — COMMENTED** [review]
R changes LGTM

**Review by smurching — COMMENTED** [review]
Proto changes also look good to me. My understanding of the changes is:

* None of the fields we're removing are required by the MLflow 0.9.1 server ([link](https://github.com/mlflow/mlflow/blob/v0.9.1/mlflow/server/handlers.py#L181)), as if they were then MLflow 1.0 clients would break against the 0.9.1 server.
* The 1.0 server will silently ignore fields we've removed here when parsing requests from 0.9.1 clients. Also, just did a sanity check that the 0.9.1 Python client does in fact set run attributes like source type as tags, so the 0.9.1 Python client will work against the 1.0 server. Unfortunately the same isn't true of the R 0.9.1 client (which you actually fixed to use tags in this PR), but I think that's ok. The Java 0.9.1 client doesn't allow setting run attributes except via the low-level `MlflowClient.createRun(Service.CreateRun request)` API, so it's a non-issue.
* The same holds true for the search API - 0.9.1 clients APIs like `MlflowClient.search_runs` use the filter field, so there won't be any behavior change for 0.9.1 clients if we stop handling `anded_expressions`.

Could be good to get another reviewer to check the proto changes though, just to be sure.

**Comment by smurching** [comment]
Actually, the 0.9.1 Python client doesn't set source type etc as tags, so we will lose that metadata from 0.9.1 clients talking to 1.0 servers (see [Python client](https://github.com/mlflow/mlflow/blob/v0.9.1/mlflow/store/rest_store.py) and [RestStore logic](https://github.com/mlflow/mlflow/blob/v0.9.1/mlflow/store/rest_store.py#L138)). But I guess that's ok?

**Comment by smurching** [comment]
^It's probably ok, since Python's `mlflow.start_run` (which is probably much more commonly-used than the lower-level `MlflowClient.create_run`) does set tags in 0.9.1 ([link](https://github.com/mlflow/mlflow/blob/v0.9.1/mlflow/tracking/fluent.py#L127))

**Comment by smurching** [comment]
For 1.0 client -> 0.9.1 server compatibility, one small issue is that the 0.9.1 server I think will drop run metadata (source type etc) set by 1.0 clients, as the 0.9.1 server / store does not read this metadata from tags when responding to GetRun, SearchRuns API requests: https://github.com/mlflow/mlflow/blob/v0.9.1/mlflow/store/file_store.py#L363

File: mlflow/server/js/src/sdk/MlflowMessages.js [hunk 1]
```diff
@@ -112,15 +112,6 @@ export const RunInfo = Immutable.Record({
   // optional INT64
```
Comment (smurching) [inline mlflow/server/js/src/sdk/MlflowMessages.js]: Were you able to autogenerate these changes from the proto? Or did you have to make them manually?

**Review by smurching — COMMENTED** [review]
JS changes LGTM

**Review by smurching — COMMENTED** [review]
Java changes LGTM

File: mlflow/store/file_store.py [hunk 2]
```diff
@@ -46,6 +45,11 @@ def _make_persisted_run_info_dict(run_info):
     # old mlflow versions to read
     run_info_dict = dict(run_info)
     run_info_dict['tags'] = []
+    run_info_dict['name'] = ''
```
Comment (smurching) [inline mlflow/store/file_store.py]: Makes sense, I think this change is necessary e.g. for MLflow projects running older MLflow versions to be able to properly deserialize the run. We should QA this pre-1.0 though :P

File: tests/tracking/test_fluent.py [hunk 3]
```diff
@@ -233,137 +232,41 @@ def test_start_run_defaults_databricks_notebook(empty_active_run_stack):
         active_run = start_run()
         MlflowClient.create_run.assert_called_once_with(
             experiment_id=mock_experiment_id,
-            run_name=None,
             tags=expected_tags
         )
         assert is_from_run(active_run, MlflowClient.create_run.return_value)
 
 
-def test_start_run_overrides(empty_active_run_stack):
```
Comment (smurching) [inline tests/tracking/test_fluent.py]: Hm why delete these tests?

**Review by smurching — COMMENTED** [review]
Python changes LGTM, besides one question about test deletion. 

Orthogonally to that, I was a little worried we weren't adding coverage that run metadata tags are set, but `test_rest_tracking` verifies that the CreateRun REST API sets tags appropriately in the File/SQL stores ([link](https://github.com/aarondav/mlflow/blob/0a9d12c06aaace21e8c7273fa26422f4275904a2/tests/tracking/test_rest_tracking.py#L234)) & `test_tracking` verifies that context tags are correctly set for `start_run` ([link](https://github.com/aarondav/mlflow/blob/0a9d12c06aaace21e8c7273fa26422f4275904a2/tests/tracking/test_tracking.py#L263)). Although I'm a little confused why MLFLOW_SOURCE_VERSION isn't set in the tracking test :P (it wasn't set before this PR either I guess). Mainly noting this stuff for future reviewers, thanks @aarondav!

File: tests/tracking/test_fluent.py (hunk 3 above)
Comment (aarondav) [reply tests/tracking/test_fluent.py]: The start_run properties are no longer overridable, as they don't exist

File: mlflow/server/js/src/sdk/MlflowMessages.js (hunk 1 above)
Comment (aarondav) [reply mlflow/server/js/src/sdk/MlflowMessages.js]: ... I made them manually, to help confirm w/the JS compilation that I had removed all references.

**Review by sueann — COMMENTED** [review]
looked at the proto changes and had just one comment - i'll look at python in a bit

File: mlflow/store/file_store.py [hunk 4]
```diff
@@ -360,10 +360,6 @@ def create_run(self, experiment_id, user_id, run_name, source_type,
         mkdir(run_dir, FileStore.ARTIFACTS_FOLDER_NAME)
         for tag in tags:
             self.set_tag(run_uuid, tag)
-        if parent_run_id:
-            self.set_tag(run_uuid, RunTag(key=MLFLOW_PARENT_RUN_ID, value=parent_run_id))
-        if run_name:
-            self.set_tag(run_uuid, RunTag(key=MLFLOW_RUN_NAME, value=run_name))
         return Run(run_info=run_info, run_data=None)
 
     def get_run(self, run_id):
```
Comment (sueann) [inline mlflow/store/file_store.py]: so we're not supporting reading of the deprecated fields for runs saved in the old format (e.g. by returning them in tags) right?

**Review by sueann — COMMENTED** [review]
just one question for you. looks generally okay to me otherwise.

**Comment by acroz** [comment]
This change looks great and is consistent with the changes we discussed previously.

The last field on runs that stands out is the user ID, which feels like descriptive information live the source and run name, but remains part of the HTTP and programmatic APIs. Have you considered pushing this into a tag as with the other fields?

(As a side note, we did not do with with previous PRs on tags and run info as the logic to determine the user ID is in the client, not in the fluent interface as was the case for the fields changes in previous PRs. If you accept putting this info in tag, I would suggest bringing this logic out into `mlflow.tracking.context.DefaultRunContext`).

File: mlflow/store/file_store.py (hunk 4 above)
Comment (aarondav) [reply mlflow/store/file_store.py]: This is a good point -- potentially we should consider having a migration script to change the old runs to include tags.

**Comment by aarondav** [comment]
I think it might be reasonable to move userId too, via a similar deprecation strategy as the other fields. From our side, I'm not sure this is important enough to block the MLflow 1.0 release, which we hope to  code-freeze by the end of this week, and I don't think we have time on our side to execute this change.

**Comment by aarondav** [comment]
Merging this - thanks @smurching and @sueann for reviewing!!

**Pull request merged**